packages/app-mobile/components/DialogManager/hooks/useDialogControl.ts

```diff
@@ -0,0 +1,98 @@
+import * as React from 'react';
+import { Alert, Platform } from 'react-native';
+import { DialogControl, DialogType, MenuChoice, PromptButton, PromptDialogData, PromptOptions } from '../types';
+import { _ } from '@joplin/lib/locale';
+import { useMemo, useRef } from 'react';
+
+type SetPromptDialogs = React.Dispatch<React.SetStateAction<PromptDialogData[]>>;
+
+const useDialogControl = (setPromptDialogs: SetPromptDialogs) => {
+	const nextDialogIdRef = useRef(0);
+
+	const dialogControl: DialogControl = useMemo(() => {
+		const onDismiss = (dialog: PromptDialogData) => {
+			setPromptDialogs(dialogs => dialogs.filter(d => d !== dialog));
+		};
+
+		const defaultButtons = [{ text: _('OK') }];
+		const control: DialogControl = {
+			info: (message: string) => {
+				return new Promise<void>((resolve) => {
+					control.prompt(_('Info'), message, [{
+						text: _('OK'),
+						onPress: () => resolve(),
+					}]);
+				});
+			},
+			error: (message: string) => {
+				return new Promise<void>((resolve) => {
+					control.prompt(_('Error'), message, [{
+						text: _('OK'),
+						onPress: () => resolve(),
+					}]);
+				});
+			},
+			prompt: (title: string, message: string, buttons: PromptButton[] = defaultButtons, options?: PromptOptions) => {
+				// Alert.alert doesn't work on web.
+				if (Platform.OS !== 'web') {
+					// Note: Alert.alert provides a more native style on iOS.
+					Alert.alert(title, message, buttons, options);
```

Comment:
For now, this continues to use the React Native `Alert.alert` when `.prompt` is called on Android and iOS. If desired, this could be adjusted to always use the custom alert dialogs.